[Offload][AMDGPU] OffloadAPI `queue.unittests` fails

While running the Offload unittest suites, one test in `OffloadAPI/queue.unittests` fails on AMDGPU (AMD Radeon RX 7700 XT):

```
leandro@GpuLab:~/projects/github/llvm/llvm-project$ ninja -C build/runtimes/runtimes-bins check-offload-unit
ninja: Entering directory `build/runtimes/runtimes-bins'
[0/1] Running offload unittest suites
FAIL: Offload-Unit :: OffloadAPI/./queue.unittests/30/32 (14 of 295)
******************** TEST 'Offload-Unit :: OffloadAPI/./queue.unittests/30/32' FAILED ********************
Script(shard):
--
GTEST_OUTPUT=json:/home/leandro/projects/github/llvm/llvm-project/build/runtimes/runtimes-bins/offload/unittests/OffloadAPI/./queue.unittests-Offload-Unit-241795-30-32.json GTEST_SHUFFLE=0 GTEST_TOTAL_SHARDS=32 GTEST_SHARD_INDEX=30 /home/leandro/projects/github/llvm/llvm-project/build/runtimes/runtimes-bins/offload/unittests/OffloadAPI/./queue.unittests
--

Script:
--
/home/leandro/projects/github/llvm/llvm-project/build/runtimes/runtimes-bins/offload/unittests/OffloadAPI/./queue.unittests --gtest_filter=olQueryQueueTest.SuccessEmptyAsyncQueueCheckResult/AMDGPU_AMD_Radeon_RX_7700_XT_0
--
/home/leandro/projects/github/llvm/llvm-project/offload/unittests/OffloadAPI/queue/olQueryQueue.cpp:23: Failure
Value of: IsQueueWorkCompleted
  Actual: false
Expected: true


/home/leandro/projects/github/llvm/llvm-project/offload/unittests/OffloadAPI/queue/olQueryQueue.cpp:23
Value of: IsQueueWorkCompleted
  Actual: false
Expected: true



********************
********************
Failed Tests (1):
  Offload-Unit :: OffloadAPI/./queue.unittests/olQueryQueueTest/SuccessEmptyAsyncQueueCheckResult/AMDGPU_AMD_Radeon_RX_7700_XT_0


Testing Time: 38.05s

Total Discovered Tests: 577
  Skipped:   2 (0.35%)
  Passed : 574 (99.48%)
  Failed :   1 (0.17%)
FAILED: offload/test/CMakeFiles/check-offload-unit /home/leandro/projects/github/llvm/llvm-project/build/runtimes/runtimes-bins/offload/test/CMakeFiles/check-offload-unit 
cd /home/leandro/projects/github/llvm/llvm-project/build/runtimes/runtimes-bins/offload/test && /usr/bin/python3 /home/leandro/projects/github/llvm/llvm-project/build/./bin/llvm-lit -sv /home/leandro/projects/github/llvm/llvm-project/build/runtimes/runtimes-bins/offload/test/unit
ninja: build stopped: subcommand failed.
```

See the specific log for this unittest:

```
leandro@GpuLab:~/projects/github/llvm/llvm-project$ ./build/runtimes/runtimes-bins/offload/unittests/OffloadAPI/queue.unittests
Running main() from /home/leandro/projects/github/llvm/llvm-project/third-party/unittest/googletest/src/gtest_main.cc
[==========] Running 64 tests from 9 test suites.
[----------] Global test environment set-up.
[----------] 6 tests from olCreateQueueTest
[ RUN      ] olCreateQueueTest.Success/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olCreateQueueTest.Success/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olCreateQueueTest.Success/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olCreateQueueTest.Success/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olCreateQueueTest.InvalidNullHandleDevice/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olCreateQueueTest.InvalidNullHandleDevice/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olCreateQueueTest.InvalidNullHandleDevice/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olCreateQueueTest.InvalidNullHandleDevice/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olCreateQueueTest.InvalidNullPointerQueue/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olCreateQueueTest.InvalidNullPointerQueue/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olCreateQueueTest.InvalidNullPointerQueue/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olCreateQueueTest.InvalidNullPointerQueue/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[----------] 6 tests from olCreateQueueTest (0 ms total)

[----------] 2 tests from olSyncQueueTest
[ RUN      ] olSyncQueueTest.SuccessEmptyQueue/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olSyncQueueTest.SuccessEmptyQueue/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olSyncQueueTest.SuccessEmptyQueue/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olSyncQueueTest.SuccessEmptyQueue/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[----------] 2 tests from olSyncQueueTest (0 ms total)

[----------] 6 tests from olDestroyQueueTest
[ RUN      ] olDestroyQueueTest.Success/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olDestroyQueueTest.Success/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olDestroyQueueTest.Success/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olDestroyQueueTest.Success/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olDestroyQueueTest.SuccessDelayedResolution/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olDestroyQueueTest.SuccessDelayedResolution/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olDestroyQueueTest.SuccessDelayedResolution/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olDestroyQueueTest.SuccessDelayedResolution/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olDestroyQueueTest.InvalidNullHandle/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olDestroyQueueTest.InvalidNullHandle/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olDestroyQueueTest.InvalidNullHandle/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olDestroyQueueTest.InvalidNullHandle/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[----------] 6 tests from olDestroyQueueTest (0 ms total)

[----------] 14 tests from olGetQueueInfoTest
[ RUN      ] olGetQueueInfoTest.SuccessDevice/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olGetQueueInfoTest.SuccessDevice/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olGetQueueInfoTest.SuccessDevice/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olGetQueueInfoTest.SuccessDevice/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olGetQueueInfoTest.SuccessEmpty/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olGetQueueInfoTest.SuccessEmpty/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olGetQueueInfoTest.SuccessEmpty/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olGetQueueInfoTest.SuccessEmpty/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olGetQueueInfoTest.InvalidNullHandle/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olGetQueueInfoTest.InvalidNullHandle/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olGetQueueInfoTest.InvalidNullHandle/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olGetQueueInfoTest.InvalidNullHandle/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olGetQueueInfoTest.InvalidQueueInfoEnumeration/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olGetQueueInfoTest.InvalidQueueInfoEnumeration/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olGetQueueInfoTest.InvalidQueueInfoEnumeration/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olGetQueueInfoTest.InvalidQueueInfoEnumeration/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olGetQueueInfoTest.InvalidSizeZero/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olGetQueueInfoTest.InvalidSizeZero/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olGetQueueInfoTest.InvalidSizeZero/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olGetQueueInfoTest.InvalidSizeZero/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olGetQueueInfoTest.InvalidSizeSmall/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olGetQueueInfoTest.InvalidSizeSmall/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olGetQueueInfoTest.InvalidSizeSmall/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olGetQueueInfoTest.InvalidSizeSmall/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olGetQueueInfoTest.InvalidNullPointerPropValue/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olGetQueueInfoTest.InvalidNullPointerPropValue/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olGetQueueInfoTest.InvalidNullPointerPropValue/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olGetQueueInfoTest.InvalidNullPointerPropValue/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[----------] 14 tests from olGetQueueInfoTest (0 ms total)

[----------] 10 tests from olGetQueueInfoSizeTest
[ RUN      ] olGetQueueInfoSizeTest.SuccessDevice/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olGetQueueInfoSizeTest.SuccessDevice/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olGetQueueInfoSizeTest.SuccessDevice/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olGetQueueInfoSizeTest.SuccessDevice/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olGetQueueInfoSizeTest.SuccessEmpty/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olGetQueueInfoSizeTest.SuccessEmpty/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olGetQueueInfoSizeTest.SuccessEmpty/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olGetQueueInfoSizeTest.SuccessEmpty/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olGetQueueInfoSizeTest.InvalidNullHandle/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olGetQueueInfoSizeTest.InvalidNullHandle/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olGetQueueInfoSizeTest.InvalidNullHandle/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olGetQueueInfoSizeTest.InvalidNullHandle/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olGetQueueInfoSizeTest.InvalidQueueInfoEnumeration/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olGetQueueInfoSizeTest.InvalidQueueInfoEnumeration/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olGetQueueInfoSizeTest.InvalidQueueInfoEnumeration/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olGetQueueInfoSizeTest.InvalidQueueInfoEnumeration/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olGetQueueInfoSizeTest.InvalidNullPointer/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olGetQueueInfoSizeTest.InvalidNullPointer/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olGetQueueInfoSizeTest.InvalidNullPointer/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olGetQueueInfoSizeTest.InvalidNullPointer/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[----------] 10 tests from olGetQueueInfoSizeTest (0 ms total)

[----------] 12 tests from olWaitEventsTest
[ RUN      ] olWaitEventsTest.Success/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olWaitEventsTest.Success/AMDGPU_AMD_Radeon_RX_7700_XT_0 (7 ms)
[ RUN      ] olWaitEventsTest.Success/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olWaitEventsTest.Success/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olWaitEventsTest.SuccessSingleQueue/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olWaitEventsTest.SuccessSingleQueue/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olWaitEventsTest.SuccessSingleQueue/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olWaitEventsTest.SuccessSingleQueue/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olWaitEventsTest.SuccessMultipleEvents/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olWaitEventsTest.SuccessMultipleEvents/AMDGPU_AMD_Radeon_RX_7700_XT_0 (1 ms)
[ RUN      ] olWaitEventsTest.SuccessMultipleEvents/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olWaitEventsTest.SuccessMultipleEvents/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olWaitEventsTest.InvalidNullQueue/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olWaitEventsTest.InvalidNullQueue/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olWaitEventsTest.InvalidNullQueue/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olWaitEventsTest.InvalidNullQueue/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olWaitEventsTest.InvalidNullEvent/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olWaitEventsTest.InvalidNullEvent/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olWaitEventsTest.InvalidNullEvent/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olWaitEventsTest.InvalidNullEvent/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olWaitEventsTest.InvalidNullInnerEvent/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olWaitEventsTest.InvalidNullInnerEvent/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olWaitEventsTest.InvalidNullInnerEvent/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olWaitEventsTest.InvalidNullInnerEvent/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[----------] 12 tests from olWaitEventsTest (12 ms total)

[----------] 8 tests from olLaunchHostFunctionTest
[ RUN      ] olLaunchHostFunctionTest.Success/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olLaunchHostFunctionTest.Success/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olLaunchHostFunctionTest.Success/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olLaunchHostFunctionTest.Success/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olLaunchHostFunctionTest.SuccessSequence/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olLaunchHostFunctionTest.SuccessSequence/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olLaunchHostFunctionTest.SuccessSequence/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olLaunchHostFunctionTest.SuccessSequence/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olLaunchHostFunctionTest.InvalidNullCallback/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olLaunchHostFunctionTest.InvalidNullCallback/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olLaunchHostFunctionTest.InvalidNullCallback/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olLaunchHostFunctionTest.InvalidNullCallback/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olLaunchHostFunctionTest.InvalidNullQueue/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olLaunchHostFunctionTest.InvalidNullQueue/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olLaunchHostFunctionTest.InvalidNullQueue/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olLaunchHostFunctionTest.InvalidNullQueue/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[----------] 8 tests from olLaunchHostFunctionTest (0 ms total)

[----------] 2 tests from olLaunchHostFunctionKernelTest
[ RUN      ] olLaunchHostFunctionKernelTest.SuccessBlocking/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olLaunchHostFunctionKernelTest.SuccessBlocking/AMDGPU_AMD_Radeon_RX_7700_XT_0 (501 ms)
[ RUN      ] olLaunchHostFunctionKernelTest.SuccessBlocking/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olLaunchHostFunctionKernelTest.SuccessBlocking/CUDA_NVIDIA_GeForce_RTX_5070_1 (500 ms)
[----------] 2 tests from olLaunchHostFunctionKernelTest (1001 ms total)

[----------] 4 tests from olQueryQueueTest
[ RUN      ] olQueryQueueTest.SuccessEmptyAsyncQueue/AMDGPU_AMD_Radeon_RX_7700_XT_0
[       OK ] olQueryQueueTest.SuccessEmptyAsyncQueue/AMDGPU_AMD_Radeon_RX_7700_XT_0 (0 ms)
[ RUN      ] olQueryQueueTest.SuccessEmptyAsyncQueue/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olQueryQueueTest.SuccessEmptyAsyncQueue/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[ RUN      ] olQueryQueueTest.SuccessEmptyAsyncQueueCheckResult/AMDGPU_AMD_Radeon_RX_7700_XT_0
/home/leandro/projects/github/llvm/llvm-project/offload/unittests/OffloadAPI/queue/olQueryQueue.cpp:23: Failure
Value of: IsQueueWorkCompleted
  Actual: false
Expected: true

[  FAILED  ] olQueryQueueTest.SuccessEmptyAsyncQueueCheckResult/AMDGPU_AMD_Radeon_RX_7700_XT_0, where GetParam() = 40-byte object <30-A4 57-2D 46-58 00-00 70-04 65-2E 46-58 00-00 1E-00 00-00 00-00 00-00 1E-00 00-00 00-00 00-00 30-5F 58-54 5F-30 00-00> (0 ms)
[ RUN      ] olQueryQueueTest.SuccessEmptyAsyncQueueCheckResult/CUDA_NVIDIA_GeForce_RTX_5070_1
[       OK ] olQueryQueueTest.SuccessEmptyAsyncQueueCheckResult/CUDA_NVIDIA_GeForce_RTX_5070_1 (0 ms)
[----------] 4 tests from olQueryQueueTest (0 ms total)

[----------] Global test environment tear-down
[==========] 64 tests from 9 test suites ran. (1015 ms total)
[  PASSED  ] 63 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] olQueryQueueTest.SuccessEmptyAsyncQueueCheckResult/AMDGPU_AMD_Radeon_RX_7700_XT_0, where GetParam() = 40-byte object <30-A4 57-2D 46-58 00-00 70-04 65-2E 46-58 00-00 1E-00 00-00 00-00 00-00 1E-00 00-00 00-00 00-00 30-5F 58-54 5F-30 00-00>

 1 FAILED TEST
```

**Environment:**

- LLVM version: Custom build from git
  - LLVM Project Commit Hash: 1762ba885f95bc2145cb1dcc312aea6517e4f0c9c
- Operating System: Ubuntu 24.04.3 LTS (Linux 6.14.0-37-generic)

--- a/offload/liboffload/src/OffloadImpl.cpp
+++ b/offload/liboffload/src/OffloadImpl.cpp
@@ -1255,7 +1255,7 @@ Error olQueryQueue_impl(ol_queue_handle_t Queue, bool *IsQueueWorkCompleted) {
                                                      IsQueueWorkCompleted))
       return Err;
   } else if (IsQueueWorkCompleted) {
-    // No underlying queue means there's no work to complete
+    // No underlying queue means there's no work to complete.
     *IsQueueWorkCompleted = true;
   }
   return Error::success();

--- a/offload/liboffload/src/OffloadImpl.cpp
+++ b/offload/liboffload/src/OffloadImpl.cpp
@@ -1254,6 +1254,9 @@ Error olQueryQueue_impl(ol_queue_handle_t Queue, bool *IsQueueWorkCompleted) {
     if (auto Err = Queue->Device->Device->queryAsync(Queue->AsyncInfo, false,
                                                      IsQueueWorkCompleted))
       return Err;
+  } else if (IsQueueWorkCompleted) {
+    // No underlying queue means there's no work to complete
+    *IsQueueWorkCompleted = true;
   }
   return Error::success();
 }

